src/visualization/visualize.py

```diff
@@ -24,14 +23,14 @@ def graph_stock(self):
         
         plt.xlabel("Date")
         plt.ylabel("Price")
-        stock_open_price = pd.Series(self._ticker['Open'].to_numpy(), index=pd.date_range(self._ticker['Date'].iloc[0], periods=503, freq='D'))
-        stock_close_price = pd.Series(self._ticker['Close'].to_numpy(), index=pd.date_range(self._ticker['Date'].iloc[0], periods=503, freq='D'))
+        stock_open_price = pd.Series(self._ticker['Open'].to_numpy(), index=pd.date_range(self._ticker['Date'].iloc[0], periods=len(self._ticker), freq='D'))
```

Comment:
Use `freq='B'` for business days. Also, there's an parameter for `holidays` to get our dates lined up exactly.

Reply:
removed pyalgo